pyClient/test_commands/scenario.py

```diff
@@ -257,24 +262,26 @@ def charlie_withdraw(
         (output_note1, pk_charlie),
         (output_note2, pk_charlie)])
 
-    # Hash the pk_sender and cipher-texts
-    ciphers = pk_sender + ciphertexts[0] + ciphertexts[1]
-    hash_ciphers = sha256(ciphers).hexdigest()
+    # Hashing all inputs of the signature
+    # Encode the ciphertexts and ephemeral encryption key
+    data_to_be_signed = pk_sender + ciphertexts[0] + ciphertexts[1]
 
-    # Hash the proof
+    # Encode the proof
     proof: List[str] = []
     for key in proof_json.keys():
         if key != "inputs":
             proof.extend(proof_json[key])
-    hash_proof = sha256(encode_to_hash(proof)).hexdigest()
+    data_to_be_signed += encode_to_hash(proof)
+
+    # Encode the primary inputs
+    encoded_inputs = encode_to_hash(proof_json["inputs"])
+    data_to_be_signed += encoded_inputs
 
-    # Encode and hash the primary inputs
-    encoded_inputs = joinsplit.encode_pub_input_to_hash(proof_json["inputs"])
-    hash_inputs = sha256(encoded_inputs).hexdigest()
+    # Hash data_to_be_sign
+    hash_tobesign = sha256(data_to_be_signed).hexdigest()
 
     # Compute the joinSplit signature
-    joinsplit_sig = joinsplit.sign(
-        joinsplit_keypair, hash_ciphers, hash_proof, hash_inputs)
+    joinsplit_sig = joinsplit.sign(joinsplit_keypair, hash_tobesign)
```

Comment:
Isn't there a way to factorize the set of instructions above as a function? This would avoid code redundancy

Reply:
Done